### PR TITLE
enh(CEIP): update metadata roles 4 cloud

### PR DESCRIPTION
**Pull request by cbadr-ext**
## Description

update metadata roles

**Fixes** # (MON-12925)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **Refactor**
	- Improved code readability in user search functionality without altering its behavior.
- **New Features**
	- Enhanced role determination and assignment based on platform type in the API.
	- Updated license information processing and improved error handling for license-related issues.
- **Style**
	- Refined return types for UUID retrieval methods to enhance code clarity and accuracy.

### CODE REVIEW

File: centreon/www/api/class/centreon_ceip.class.php [hunk 1]
```diff
@@ -44,18 +44,33 @@ public function __construct()
...
+        }
+
+        /** @var null|\Core\Common\Infrastructure\FeatureFlags $featureFlags */
+        $featureFlags = $kernel->getContainer()->get(\Core\Common\Infrastructure\FeatureFlags::class);
+        if (null === $featureFlags) {
+            throw new LogicException('FeatureFlags not found in container');
+        }
+
+        $this->logger = $logger;
+        $this->featureFlags = $featureFlags;
```
Comment (callapa) [inline centreon/www/api/class/centreon_ceip.class.php:73]: Why did you replace this code when it was already working properly?
Please undo your changes.

File: centreon/www/api/class/centreon_ceip.class.php [hunk 2]
```diff
@@ -228,7 +274,7 @@ private function getLicenseInformation(): array
             }
         } catch (\Pimple\Exception\UnknownIdentifierException) {
             // The licence does not exist, 99.99% chance we are on Open source. No need to log.
-        } catch (\Exception $exception) {
+        } catch (\Exception | \Throwable $exception) {
```
Comment (callapa) [inline centreon/www/api/class/centreon_ceip.class.php:277]: Throwable is an interface, while Exception is a class that implements Throwable.
Throwable is therefore sufficient.

```suggestion
        } catch (\Throwable $exception) {
```

File: centreon/www/class/centreonUUID.class.php [hunk 3]
```diff
@@ -54,7 +54,7 @@ public function __construct($db)
     /**
      * Get Centreon UUID
      *
-     * @return bool|string
+     * @return false|string
```
Comment (callapa) [inline centreon/www/class/centreonUUID.class.php:57]:
```suggestion
     * @return string
```

File: centreon/www/api/class/centreon_ceip.class.php [hunk 4]
```diff
@@ -144,7 +161,25 @@ private function getVisitorInformation(): array
...
+            INNER JOIN acl_group_actions_relations AS agar ON (agar.acl_action_id = aar.acl_action_rule_id)
+            INNER JOIN acl_group_topology_relations AS agtr ON (agtr.acl_group_id = agar.acl_group_id)
+            INNER JOIN acl_topology AS atp ON (atp.acl_topo_id = agtr.acl_topology_id)
+            INNER JOIN acl_topology_relations AS atr ON (atr.acl_topo_id = atp.acl_topo_id)
+            INNER JOIN topology AS t ON (t.topology_id = atr.topology_topology_id)
+            WHERE t.topology_page = '601' --'601' is the ID of the page for monitoring configuration
+            SQL
+            );
+            if ($countAcl > 0) {
+                $role = 'editor';
```
Comment (callapa) [inline centreon/www/api/class/centreon_ceip.class.php:182]: Editor Cloud's role has nothing to do with ACLs

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/www/api/class/centreon_ceip.class.php [hunk 5]
```diff
@@ -124,6 +125,20 @@ private function getVisitorInformation(): array
             ? 'admin'
             : 'user';
 
+        $countAcl = (int) $this->sqlFetchValue(
+            <<<SQL
+            SELECT COUNT(*) AS countAcl
+            FROM acl_actions_rules AS aar
```
Comment (lpinsivy) [inline centreon/www/api/class/centreon_ceip.class.php:131]: Why you use acl_actions_rules table while we need topology and where is the relation with contact_id or contact groups of the user?

**Review by lpinsivy — CHANGES_REQUESTED** [review]

File: centreon/www/api/class/centreon_ceip.class.php [hunk 6]
```diff
@@ -120,31 +121,28 @@ private function getVisitorInformation(): array
...
+            $grouplistStr = $this->user->access->getAccessGroupsString('NAME');
+
+            // Check main ACL Group
+            if (preg_match('/customer_admin_acl/', $grouplistStr)) {
+                $role = 'Administrator';
+            } elseif (preg_match('/customer_editor_acl/', $grouplistStr)) {
+                $role = 'Editor';
+            } elseif (preg_match('/customer_user_acl/', $grouplistStr)) {
+                $role = 'User';
+            } else {
```
[the remark this answers is not in the document]
Comment (lpinsivy) [reply centreon/www/api/class/centreon_ceip.class.php:135]: @a-launois why to add another space?

[the remark this answers is not in the document]
Comment (a-launois) [reply centreon/www/api/class/centreon_ceip.class.php:138]: It's not about the space but about the lines removed
I edited the suggestion to remove the space

**Review by a-launois — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**